File: 26.Remove_Duplicates_from_Sorted_Array/java/Remove_Duplicates_from_Sorted_Array.java
public class Remove_Duplicates_from_Sorted_Array {
    public static int removeDuplicates(int[] nums) {
        int l = 0, r = 1;
        while (r < nums.length) {
            if (nums[l] != nums[r]) {
                l++;
                if (l != r + 1) {
                    nums[l] = nums[r];
                }
            }
            r++;
        }
        return l + 1;
    }

    public static void main(String[] args) {
        int[] nums = { 1, 1, 2, 2, 2, 3 };

        int k = removeDuplicates(nums);

        System.out.println("k = " + k);
        System.out.print("nums = [");
        for (int i = 0; i < k; i++) {
            if (i > 0) {
                System.out.print(", ");
            }
            System.out.print(nums[i]);
        }
        System.out.println("]");
    }
}

File: Easy/26.Remove_Duplicates_from_Sorted_Array/javascript/Remove_Duplicates_from_Sorted_Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let left = 0
  let right = 1
  while (right < nums.length) {
    if (nums[left] !== nums[right]) {
      left++
      nums[left] = nums[right]
    }
    right++
  }
  return left + 1
}

const nums = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]
console.log(removeDuplicates(nums))
